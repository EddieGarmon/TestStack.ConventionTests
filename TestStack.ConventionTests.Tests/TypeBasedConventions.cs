namespace TestStack.ConventionTests.Tests
{
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using NUnit.Framework;
    using TestAssembly;
    using TestStack.ConventionTests.Conventions;

    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class TypeBasedConventions
    {
        [Test]
        public void Test()
        {
            var types = typeof(ClassWithNonVirtualMethod).Assembly.GetTypes();

            var exception = Assert.Throws<ConventionFailedException>(() => Convention.Is<AllMethodsAreVirtual>(types));

            Approvals.Verify(exception.Message);
        }
    }
}