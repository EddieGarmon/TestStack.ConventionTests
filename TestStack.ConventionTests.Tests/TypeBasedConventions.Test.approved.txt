Convention has failing items
	TestAssembly.ClassWithNonVirtualMethod has non virtual method(s):
	NonVirtual
